### PR TITLE
feat: app build version for latest app

### PR DESCRIPTION
**Pull request by Ikalli**
# PR Template

## Checklist

- [x] E2E finished.
- [ ] Local, Self Hosted and Development environment test finished.
- [ ] Local, E2E, Self Hosted, Development and Production env variable checked.
- [x] (Backend only) DB migration file creation checked.
- [x] **If this is a API deleted or changed**: Check side effects for API deletion or related code change.

## Releated issues
#426

### CODE REVIEW

File: dogu-actions/prepare/src/main.ts [hunk 1]
```diff
@@ -34,10 +24,38 @@ ActionKit.run(async ({ options, logger, input, deviceHostClient, consoleActionCl
...
+  const resolvedAppVersion = appVersion || process.env.DOGU_APP_VERSION || '';
+  const resolvedAppPackageName = appPackageName || process.env.DOGU_APP_PACKAGE_NAME || '';
+
+  if (resolvedAppPackageName) {
+    const currentPlatformPackageName =
+      typeof resolvedAppPackageName === 'object'
+        ? (() => {
+            const platformAppVersion = Reflect.get(resolvedAppPackageName, DOGU_DEVICE_PLATFORM) as string | undefined;
+            if (!platformAppVersion) {
+              throw new Error(`Invalid app version: ${stringify(resolvedAppPackageName)} for platform: ${DOGU_DEVICE_PLATFORM}`);
```
Comment (yowpark) [inline dogu-actions/prepare/src/main.ts:39]: Is the error message incorrect?. I think it should be Invalid app package name: ...

File: dogu-actions/run-test/src/main.ts [hunk 2]
```diff
@@ -52,7 +53,21 @@ ActionKit.run(async ({ options, logger, input, deviceHostClient, consoleActionCl
...
-  if (resolvedAppVersion) {
+  const resolvedAppPackageName = appPackageName || process.env.DOGU_APP_PACKAGE_NAME || '';
+
+  if (resolvedAppPackageName) {
+    const currentPlatformPackageName =
+      typeof resolvedAppPackageName === 'object'
+        ? (() => {
+            const platformAppVersion = Reflect.get(resolvedAppPackageName, DOGU_DEVICE_PLATFORM) as string | undefined;
+            if (!platformAppVersion) {
+              throw new Error(`Invalid app version: ${stringify(resolvedAppPackageName)} for platform: ${DOGU_DEVICE_PLATFORM}`);
```
Comment (yowpark) [inline dogu-actions/run-test/src/main.ts:64]: Is the error message incorrect?. I think it should be "Invalid app package name: ..."

File: packages/typescript/action-kit/src/built-in-actions/download-app.ts [hunk 3]
```diff
@@ -28,28 +28,33 @@ export async function downloadApp(
...
   if (applications.length === 0) {
-    throw new Error(`No application found for version ${appVersion} and extension ${extension}`);
+    throw new Error(
+      `No application found for ${
+        by.appPackageName ? `appPackageName ${by.appPackageName}` : by.appVersion ? `appVersion ${by.appVersion}` : 'empty condition'
+      } and extension ${extension}`,
+    );
   }
+
   const application = applications[0];
```
Comment (yowpark) [inline packages/typescript/action-kit/src/built-in-actions/download-app.ts:58]: Shouldn't we filter by.appPackageName among the app list obtained from consoleActionClient.getApplicationsWithUniquePackage?

File: projects/console-web-server/src/module/routine/pipeline/pipeline.common.ts [hunk 4]
```diff
@@ -67,6 +72,21 @@ export function parseAppVersion(appVersionRaw: JobSchema['appVersion'], platform
   return null;
 }
 
+export function parseAppPackageName(appPackageNameRaw: JobSchema['appPackageName'], platform: Platform): string | null {
+  if (appPackageNameRaw) {
+    if (typeof appPackageNameRaw === 'string') {
+      return appPackageNameRaw;
+    } else if (typeof appPackageNameRaw === 'object') {
+      const platformType = platformTypeFromPlatform(platform);
+      const appVersion = appPackageNameRaw[platformType];
```
Comment (yowpark) [inline projects/console-web-server/src/module/routine/pipeline/pipeline.common.ts:81]: I think this variable name should be appPackageName.

**Comment by oneofthezombies** [comment]
LGTM!

File: packages/typescript/action-kit/src/built-in-actions/download-app.ts (hunk 3 above)
Comment (Ikalli) [reply packages/typescript/action-kit/src/built-in-actions/download-app.ts]: 👍

**Pull request merged**